extensions/amp-viewer-integration/0.1/messaging.js

```diff
@@ -68,121 +86,141 @@ export class Messaging {
+    const requestId = ++this.requestIdCounter_;
     let promise = undefined;
     if (awaitResponse) {
       promise = new Promise((resolve, reject) => {
         this.waitingForResponse_[requestId] = {resolve, reject};
       });
     }
-    this.sendMessage_(requestSentinel_, requestId, eventType,
-      payload, awaitResponse);
+    this.sendMessage_(requestId, MessageType_.REQUEST, eventType, data,
+      awaitResponse, null);
     return promise;
   }
 
   /**
    * I'm responding to a request that Bob made earlier.
-   * @param {string} requestId
-   * @param {*} payload
+   * @param {number} requestId
+   * @param {string} name
+   * @param {*} data
    * @private
    */
-  sendResponse_(requestId, payload) {
+  sendResponse_(requestId, name, data) {
     dev().info(TAG, 'messaging.js -> sendResponse_');
     this.sendMessage_(
-      responseSentinel_, requestId, null, payload, false);
+      requestId, MessageType_.RESPONSE, name, data, null, null);
   }
 
   /**
-   * @param {string} sentinel
-   * @param {string} requestId
-   * @param {string|null} eventType
-   * @param {*} payload
-   * @param {boolean} awaitResponse
+   * @param {number} mId
+   * @param {string} mType
+   * @param {string} mName
+   * @param {*} mData
+   * @param {?boolean} mRsvp
+   * @param {*} mErr
    * @private
    */
-  sendMessage_(sentinel, requestId, eventType, payload, awaitResponse) {
+  sendMessage_(mId, mType, mName, mData, mRsvp, mErr) {
+    /** @type {Message} */
     const message = {
-      sentinel,
-      requestId,
-      type: eventType,
-      payload,
-      rsvp: awaitResponse,
+      app: APP,
+      requestid: mId,
+      type: mType,
+      name: mName,
+      data: mData,
+      rsvp: mRsvp,
+      error: mErr,
     };
     this.target_./*OK*/postMessage(message, this.targetOrigin_);
   }
 
   /**
-   * @param {string} requestId
+   * @param {number} requestId
+   * @param {string} name
    * @param {*} reason
    * @private
    */
-  sendResponseError_(requestId, reason) {
+  sendResponseError_(requestId, name, reason) {
     this.sendMessage_(
-      responseSentinel_, requestId, 'ERROR', reason, false);
+      requestId, MessageType_.RESPONSE, name, null, null, reason);
   }
 
   /**
    * I'm handing an incoming request from Bob. I'll either respond normally
    * (ex: "got it Bob!") or with an error (ex: "I didn't get a word of what
    * you said!").
-   * @param {*} message
+   * @param {Message} message
    * @private
    */
   handleRequest_(message) {
-    dev().info(TAG, 'messaging.js -> handleRequest_');
-    const requestId = message.requestId;
-    const promise = this.requestProcessor_(message.type, message.payload,
-        message.rsvp);
+    dev().info(TAG, 'messaging.js -> handleRequest_', message);
+    dev().assert(this.requestProcessor_,
+      'Cannot handle request because handshake is not yet confirmed!');
+    const requestId = message.requestid;
+    const msg = dev().assertString(message.name);
```

Comment:
yes, setRequestProcessor needs the assert. 
sendResponse & sendResponseError need name to be a {!string} type because the message type for name is {!string}